src/main/java/org/jenkinsci/plugins/GithubSecurityRealm.java

```diff
@@ -663,6 +662,31 @@ public GroupDetails loadGroupByGroupname(String groupName)
         }
     }
 
+    /*
+       Migrate settings from 0.20 to 0.21+
+     */
+    @Extension
+    public static final class Migrator extends ItemListener {
+        @SuppressWarnings("deprecation")
+        @Override
+        public void onLoaded() {
+            try {
+                Jenkins instance = Jenkins.getInstance();
+                if(instance.getSecurityRealm() instanceof GithubSecurityRealm) {
+                    GithubSecurityRealm myRealm = (GithubSecurityRealm) instance.getSecurityRealm();
+                    if(myRealm.getOauthScopes() == null || myRealm.getOauthScopes() == "") {
```

Comment:
I'm not sure about the empty string test.  I think the only way the `GithubSecurityRealm.oauthScopes` field would be `""` is if the user had already set it?